local-interchaintest/examples/ibc_transfer_osmo_juno_gaia_pfm.rs

```diff
@@ -0,0 +1,296 @@
+    info!("Start input balance: {:?}", start_input_balance);
+
+    // We need a normal account on Gaia to be the output account for the IBC transfer service
+    let output_account = test_ctx.get_chain(GAIA_CHAIN_NAME).admin_addr.clone();
+    info!("Output account: {:?}", output_account);
+
+    let start_output_balance = bank::get_balance(
+        test_ctx
+            .get_request_builder()
+            .get_request_builder(GAIA_CHAIN_NAME),
+        &output_account,
+    )
+    .iter()
+    .find(|bal| bal.denom == GAIA_CHAIN_DENOM)
+    .map_or(0, |bal| bal.amount.u128());
+    info!("Start output balance: {:?}", start_output_balance);
+
+    info!("Prepare the IBC transfer service contract");
+    let ibc_transfer_svc_contract_path = format!(
+        "{}/artifacts/valence_generic_ibc_transfer_service.wasm",
+        current_dir.display()
+    );
+
+    let mut uploader = test_ctx.build_tx_upload_contracts();
+    uploader
+        .with_chain_name(OSMOSIS_CHAIN_NAME)
+        .send_single_contract(&ibc_transfer_svc_contract_path)?;
+
+    let code_id_ibc_transfer_svc = *test_ctx
+        .get_chain(OSMOSIS_CHAIN_NAME)
+        .contract_codes
+        .get("valence_generic_ibc_transfer_service")
+        .unwrap();
+
+    info!("Creating IBC transfer service contract");
+    let transfer_amount = 1_000_000_000_000u128;
+    let osmo_juno_path = &(OSMOSIS_CHAIN_NAME.to_string(), JUNO_CHAIN_NAME.to_string());
+    let ibc_transfer_instantiate_msg = valence_service_utils::msg::InstantiateMsg::<ServiceConfig> {
+        owner: OSMOSIS_CHAIN_ADMIN_ADDR.to_string(),
+        processor: OSMOSIS_CHAIN_ADMIN_ADDR.to_string(),
+        config: ServiceConfig::with_pfm_map(
+            ServiceAccountType::Addr(input_account.clone()),
+            output_account.clone(),
+            UncheckedDenom::Native(atom_on_osmo_via_juno.clone()),
+            IbcTransferAmount::FixedAmount(transfer_amount.into()),
+            "".to_owned(),
+            valence_neutron_ibc_transfer_service::msg::RemoteChainInfo {
+                channel_id: test_ctx
+                    .transfer_channel_ids
+                    .get(osmo_juno_path)
+                    .unwrap()
+                    .clone(),
+                ibc_transfer_timeout: Some(600u64.into()),
+            },
+            BTreeMap::from([(
+                atom_on_osmo_via_juno.clone(),
+                PacketForwardMiddlewareConfig {
+                    local_to_hop_chain_channel_id: test_ctx
+                        .get_transfer_channels()
+                        .src(OSMOSIS_CHAIN_NAME)
+                        .dest(JUNO_CHAIN_NAME)
+                        .get(),
+                    hop_to_destination_chain_channel_id: test_ctx
+                        .get_transfer_channels()
+                        .src(JUNO_CHAIN_NAME)
+                        .dest(GAIA_CHAIN_NAME)
+                        .get(),
+                    hop_chain_receiver_address: JUNO_CHAIN_ADMIN_ADDR.to_string(),
```

Comment:
for debugging I'd start with setting this to an invalid bech32 address so that the funds don't settle in this addr in case of a failed pfm

Reply:
Test is now passing .. it was due to the (incorrect) use of an old version (non-cosmwasm 2.0) Osmosis chain.